### DISCUSSION TITLE
Watching?

### DISCUSSION THREAD

**Discussion by alecmev** [paularmstrong/onerepo#670] [Q&A] [answered]
I see that the [Jest plugin](https://onerepo.tools/plugins/jest/) has a `--watch` option, so I understand that long-running processes aren't an after-thought, but what would be a recipe for, say, a mainstream example, [`next dev`](https://nextjs.org/docs/app/api-reference/next-cli#development)? And `tsc --watch` of the whole repo? And running them in parallel, while also putting changed files through ESLint and CSpell? Would I need to [write commands](https://onerepo.tools/docs/commands/) for these things? I don't mind doing that per se, just wondering what's the blessed way.

I'm currently maintaining a monorepo without an orchestrator (shopping for one), and I think this is the biggest pain point. The only thing I can reliably do is `tsc --watch`, everything else - ESLint, CSpell, Prettier, Knip, Yarn constraints - is done either through editor integrations or on-demand, over the whole repo.

**Comment by paularmstrong** [comment, marked as answer]
Writing commands is the most recommended way to do this, as we've found more often than not that we need to include default flags for third-party commands, or run extra setup or checks under certain conditions, and it's better to have a wrapper command to handle this. These wrappers can start pretty simple. Here's an example that runs the netlify dev server: https://github.com/paularmstrong/paularmstrong.dev/blob/64049b3229292b99ce4c007c96cdae321d0c7d2c/commands/start.ts#L12-L20


There's an alternative to bootstrap and get things running for the simplest cases: [passthrough commands](https://onerepo.tools/core/workspace/#passthrough-commands). These are discouraged mostly due to the previous reasoning and because you are limited in help documentation and extra options/arguments that can be passed through. This is exactly how oneRepo runs documentation with Astro: https://github.com/paularmstrong/onerepo/blob/b797bba2efdd4b5cbde9103825d4b7f75da46e1a/docs/onerepo.config.ts#L5-L10
